### PR TITLE
Documentation for Users endpoint related objects

### PR DESCRIPTION
**Pull request by nanaya**
Todo:
- [x] Beatmapset object
- [x] Score object
- [x] User object (was halfway done whoops)
- [x] <del>Checks and probably add more details</del> to be done later
- [x] depends on #6585

### CODE REVIEW

File: resources/docs/source/includes/_structures.md [hunk 1]
```diff
@@ -305,6 +305,157 @@ The values of the cursor should be provided to next request of the same endpoint
...
+
+Field     | Type
+----------|-------------
+mode      | GameMode
+beatmap   | EventBeatmap
+user      | EventUser
+
+#### userSupportAgain
+
+When a user becomes supporter for the second or more time.
```
Comment (notbakaneko) [inline resources/docs/source/includes/_structures.md:431]:
```suggestion
When a user supports osu! for the second time and onwards.
```

File: resources/docs/source/includes/_structures.md [hunk 2]
```diff
@@ -305,6 +305,157 @@ The values of the cursor should be provided to next request of the same endpoint
...
+
+Field     | Type
+----------|----------
+user      | EventUser
+
+#### userSupportFirst
+
+When a user becomes supporter for the first time.
```
Comment (notbakaneko) [inline resources/docs/source/includes/_structures.md:439]:
```suggestion
When a user becomes a supporter for the first time.
```

File: resources/docs/source/includes/_structures.md [hunk 3]
```diff
@@ -305,6 +305,157 @@ The values of the cursor should be provided to next request of the same endpoint
...
+
+Field     | Type
+----------|----------
+user      | EventUser
+#### userSupportGift
+
+When a user received supporter tag gift from another user.
```
Comment (notbakaneko) [inline resources/docs/source/includes/_structures.md:446]:
```suggestion
When a user if gifted a supporter tag from another user.
```

**Review by notbakaneko — CHANGES_REQUESTED** [review]
I wonder if `supporter` should be changed to `osu!supporter` to match the wiki 🤔

**Review by notbakaneko — APPROVED** [review]

**Comment by nanaya** [comment]
wat ಠ_ಠ this thing isn't done yet

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**